src/ppx_repr/lib/engine.ml

```diff
@@ -22,6 +22,8 @@ let map_lident f = function
   | Ldot (l, s) -> Ldot (l, f s)
   | Lident s -> Lident (f s)
 
+let repr_name_of_type_name = function "t" -> "t" | x -> x ^ "_t"
```

Comment:
Regarding this, I was wondering if it would be possible, instead of deciding that `t` is the _default_ type to add a `~main` option that tells the deriver we don't want representatives and functions associated to this type to be prefixed or suffixed. We would enforce the existence of a single `main` type in a module and give more control over what the deriver will produce.

Reply:
Good point. I'd like to have some particular use-case in mind before generalising in that way, but in principle it sounds worthwhile.